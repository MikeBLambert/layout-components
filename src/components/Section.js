import Component from './Component';
import styles from './Section.css';

export default class Section extends Component {
  render() {
    const sectionTitles = this.props.sectionTitles.map(title => {
      return `<h1>${title}</h1>`;
    });

    return `
    <section>
      <div>${sectionTitles.join(' ')}</div>
    </section>
    `;
  }
}
