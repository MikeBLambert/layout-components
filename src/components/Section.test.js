import Section from './Section';

describe('Section Component', () => {

  it('should be a section', () =>{
    const section = new Section({
      sectionTitles: ['About', 'Portfolio', 'Blog', 'Contact']
    });
    expect(section.render()).toMatchSnapshot();
  });
});
