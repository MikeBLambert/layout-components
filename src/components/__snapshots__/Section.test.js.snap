// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Component should be a section 1`] = `
"
    <section>
      <div><h1>About</h1> <h1>Portfolio</h1> <h1>Blog</h1> <h1>Contact</h1></div>
    </section>
    "
`;
